doc/manual/cli.md

```diff
@@ -34,12 +34,96 @@ Arguments:
 This document contains some more in-depth documentation on some of the CLI
 commands.
 
+## Customize mode: passing parameters to configurations
+
+**Warning: the CLI customize mode is still experimental and subject to breaking
+changes, although it has been there for several minor versions and isn't
+expected to radically change.**
+
+The customize mode of the CLI allows to pass values to a Nickel configuration
+directly from the command line. It is supported by the `nickel eval` and `nickel
+export` commands. After providing the files to evaluate (and potential other
+arguments), you can provide a list of key-value assignments that set or override
+the top-level fields of the configuration after the `--` separator.
+
+For example, for a file `main.ncl`:
+
+```nickel
+{
+  params | not_exported = {
+    environment | [| 'dev,  'prod |],
+    mock_db | Bool,
+  },
+  test_string = "env=${params.environment}, mock_db=${params.mock_db}",
+}
+```
+
+You can run:
+
+```console
+$ nickel export main.ncl -- params.environment=\'dev params.mock_db=true
+{
+  "test_string": "env=dev, mock_db=true"
+}
+~
+```
+
+The customize mode only allow to set fields that are not already set in the
+configuration (or only have a defaut value). You can override existing values as
+well, but you need to use `--override <key=value>`:
+
+```console
+$ nickel export main.ncl -- params.environment=\'dev params.mock_db=true --override 'test_string="my own test string, eventually"'
+{
+  "test_string": "my own test string, eventually"
+}
+```
+
+The value might be any valid Nickel expression. Note that strings or other
+Nickel characters that have a special meaning in the shell need to be properly
+escaped.
+
+For a given configuration, you can list the fields available for assignment or
+overriding with `list`:
+
+```console
+$ nickel export main.ncl -- list
+Input fields:
+- params.environment: <[| 'dev, 'prod |]>
+- params.mock_db: <Bool>
+
+Overridable fields (require `--override`):
+- test_string
+
+Use the `query` subcommand to print a detailed description of a specific field. See `nickel help query`.
+```
+
+Detailed help on the customize mode is available with `nickel export main.ncl --
+help`.
+
+### Sigil expressions
+
+The value part of an assignment supports an extended syntax called sigil
+expressions, of the form `@<selector>[/attribute]:<argument>`. Currently, the
+only supported selector is `env`, which fetches an environment variable and puts
+its value in the corresponding field as an expression:
+
+```console
+nickel export myconfig.ncl -- environment.path=@env:PATH enviornment.classpath=@env:CLASSPATH
+```
+
+`@env` doesn't support any attribute and fails if the environment variable isn't
+set.
```

Comment:
Maybe it's worth pointing out that this is sort of the same as `environment.path=$PATH` except with sane semantics regarding escaping and undefined variables